api/v1alpha1/perconaservermysql_types.go

```diff
@@ -44,6 +44,9 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // PerconaServerMySQLSpec defines the desired state of PerconaServerMySQL
+// +kubebuilder:validation:XValidation:rule="!(self.mysql.clusterType == 'async') || self.unsafeFlags.orchestrator || self.orchestrator.enabled",message="Invalid configuration: When 'mysql.clusterType' is set to 'async', 'orchestrator.enabled' must be true unless 'unsafeFlags.orchestrator' is enabled"
+// +kubebuilder:validation:XValidation:rule="!(self.mysql.clusterType == 'async') || self.unsafeFlags.proxy || self.proxy.haproxy.enabled",message="Invalid configuration: When 'mysql.clusterType' is set to 'async', 'proxy.haproxy.enabled' must be true unless 'unsafeFlags.proxy' is enabled"
+// +kubebuilder:validation:XValidation:rule="!(self.mysql.clusterType == 'async') || self.proxy.router == null || !has(self.proxy.router.enabled) || !self.proxy.router.enabled",message="Invalid configuration: When 'mysql.clusterType' is set to 'async', 'proxy.router.enabled' must be disabled"
```

Comment:
We could introduce more validations e.g. for the cluster sizes, but adding them now means that we will start effecting the unit testing we have like

```
var _ = Describe("Unsafe configurations", Ordered, func() {
	ctx := context.Background()

```

This could also mean that the old validations will become obsolete. We will keep the focus for now only on the cluster type as we initially discussed, and followup with adding more rules here.